Add schema definition scripts to code repository

zednis opened the pull request:
changes
- initial commit of schema definition statements

closes https://github.com/zednis/alexa-skill-meeting-scheduler/issues/36

zednis commented:
@mwang25277, I used DataGrip IDE to generate this schema DDL from the existing DB table definitions.  Does it look correct to you?

mwang25277 commented on schema/schema.sql:
```diff
@@ -0,0 +1,42 @@
+CREATE TABLE IF NOT EXISTS Calendar (
+	id INT AUTO_INCREMENT PRIMARY KEY,
+	user_fk INT NULL,
```
I don't think the user_fk field is necessary anymore, since the User table has a primary_calendar_fk field

zednis replied:
ok, I will drop.

mwang25277 commented on schema/schema.sql:
```diff
@@ -0,0 +1,42 @@
+	createdAt datetime default CURRENT_TIMESTAMP NOT NULL,
+	updatedAt datetime default CURRENT_TIMESTAMP NOT NULL on update CURRENT_TIMESTAMP,
+	user_calendar_fk INT NULL,
+	organizing_event INT NULL,
```
Maybe both user_calendar_fk and organizing_event should be NOT NULL

zednis replied:
let's leave organizing_event as nullable for now else every meeting, even ones that are organizing events, will need to have a non-Null FK value there.

I agree that user_calendar_fk should be non-null though, so I will update.

zednis commented:
@mwang25277 , @omerosman 

I made some updates to the schema:
- field names all now use lower_case convention (was previously using a mix of camelCase and lower_case)
- removed _fk from field names and added to foreign key constraint names
- removed Calendar.user_fk (we weren't currently using it)
- added "ON DELETE CASCADE" to foreign keys that reference calendars.  This means we will delete a user, room, or meeting if we delete the associated calendar
- added "ON DELETE CASCADE" to meeting to organizing meeting foreign key.  This means that if we delete the organizing event it should delete all child events (events on non-organizer participant calendars and the meeting room calendar)

My hope is that the ON DELETE CASCADE constraints will make keeping the database consistent when doing deletes simpler.

zednis commented:
@omerosman you should be able to execute this script to setup a local DB instance, but you will then have to make some small updates to your local copy of the mysql code to handle a few field name updates.

zednis commented:
@mwang25277 Do you think this PR is ready to merge?  With schema updates in current and upcoming tickets I would like this schema file to be available so the tickets can document schema changes in their PRs.

zednis commented:
Per discussion on slack @mwang25277 is good with merging the pull request.
